<launch>
  <node pkg="mlcc" type="pose_refine" name="pose_refine" output="screen">
    <!-- test scene -->
    <param name="data_path" type="string" value="$(find mlcc)/scene2/"/>

    <!-- optimization iteration number -->
    <param name="max_iter" type="int" value="10"/>

    <!-- base LiDAR: 0-2 is the MID from left to right, 3 is AVIA -->
    <param name="base_lidar" type="int" value="3"/>

    <!-- largest voxel size:
    if AVIA is set as base LiDAR use 4, if MID is set as base LiDAR use 4/2 for scene-1/scene-2 -->
    <param name="voxel_size" type="double" value="4"/>

    <!-- base LiDAR downsampling size:
    downsample_base: if AVIA is set as base LiDAR use 0.4, if MID is set as base LiDAR use 0.1
    eigen_threshold: if AVIA is set as base LiDAR use 10, if MID is set as base LiDAR use 20 -->
    <param name="downsample_base" type="double" value="0.4"/>
    <param name="load_original" type="bool" value="1"/>
    <param name="eigen_threshold" type="double" value="10"/>
  </node>

  <arg name="rviz" default="true" />
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"
    args="-d $(find mlcc)/rviz_cfg/rosbag.rviz" />
  </group>
</launch>